management-api-server/src/main/java/com/datastax/mgmtapi/resources/models/BaseEntity.java

```diff
@@ -0,0 +1,213 @@
+      @JsonProperty("annotations") List<String> annotations,
+      @JsonProperty("mediaType") MediaType mediaType,
+      @JsonProperty("language") String language,
+      @JsonProperty("encoding") String encoding) {
+    this.variant = variant;
+    this.annotations = annotations;
+    this.mediaType = mediaType;
+    this.language = language;
+    this.encoding = encoding;
+  }
+
+  @Override
+  public int hashCode() {
+    int hash = 5;
+    hash = 83 * hash + Objects.hashCode(this.variant);
+    hash = 83 * hash + Objects.hashCode(this.annotations);
+    hash = 83 * hash + Objects.hashCode(this.mediaType);
+    hash = 83 * hash + Objects.hashCode(this.language);
+    hash = 83 * hash + Objects.hashCode(this.encoding);
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null) {
+      return false;
+    }
+    if (getClass() != obj.getClass()) {
+      return false;
+    }
+    final BaseEntity other = (BaseEntity) obj;
+    if (!Objects.equals(this.variant, other.variant)) {
+      return false;
+    }
+    if (!Objects.equals(this.annotations, other.annotations)) {
+      return false;
+    }
+    if (!Objects.equals(this.mediaType, other.mediaType)) {
+      return false;
+    }
+    if (!Objects.equals(this.language, other.language)) {
+      return false;
+    }
+    return Objects.equals(this.encoding, other.encoding);
+  }
+
+  public static class MediaType {
+
+    @JsonProperty(value = "type", required = false)
+    public final String type;
+
+    @JsonProperty(value = "subtype", required = false)
+    public final String subtype;
+
+    @JsonProperty(value = "parameters", required = false)
+    public final Map<String, String> parameters;
+
+    @JsonProperty(value = "wildcardType", required = false)
+    public final String wildcardType;
+
+    @JsonProperty(value = "wildcardSubtype", required = false)
+    public final String wildcardSubtype;
+
+    @JsonCreator
+    public MediaType(
+        @JsonProperty("type") String type,
+        @JsonProperty("subtype") String subtype,
+        @JsonProperty("parameters") Map<String, String> parameters,
+        @JsonProperty("wildcardType") String wildcardType,
+        @JsonProperty("wildcardSubtype") String wildcardSubtype) {
+      this.type = type;
+      this.subtype = subtype;
+      this.parameters = parameters;
+      this.wildcardType = wildcardType;
+      this.wildcardSubtype = wildcardSubtype;
+    }
+
+    @Override
+    public int hashCode() {
+      int hash = 5;
+      hash = 83 * hash + Objects.hashCode(this.type);
+      hash = 83 * hash + Objects.hashCode(this.subtype);
+      hash = 83 * hash + Objects.hashCode(this.parameters);
+      hash = 83 * hash + Objects.hashCode(this.wildcardType);
+      hash = 83 * hash + Objects.hashCode(this.wildcardSubtype);
+      return hash;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
Should we add a refactoring ticket to adopt Lombok into this project at some point?

Reply:
Please no, isn't Lombok already abandoned since most of it's benefits are in newer JDK?

Reply:
Maybe, but we aren't on a newer JDK...

Reply:
I am not in favor of Lombok. If you want to add the ticket, I won't stop you :wink:

Reply:
OK, no Lombok... Let's leave it :)

Reply:
@Miles-Garnsey, don't forget to follow our [PR comments convention](https://docs.k8ssandra.io/contribute/pr_review/#comments-convention), thanks!